core/models/configuration_template.rb

```diff
@@ -40,13 +40,22 @@ def extract_node_configurations
   end
 
   # Method analyses the structure of the template
-  # @returns [Symbol] type of the template: vagrant or docker
+  # @returns [Symbol] type of the template: vagrant, docker or aws
   def determine_template_type
     target_boxes = @node_configurations.map { |_, node| node['box'] }
-    if target_boxes.include?('docker')
-      :docker
-    else
-      :vagrant
+    template_type = nil
+    target_boxes.each do |box|
+      box_provider = %w[docker aws libvirt vbox].detect { |provider| box.include?(provider) }
+      raise("Unknown provider #{box_provider}") if box_provider.nil?
```

Comment:
Will the end user see an error message or a stack trace? User must not see the stack trace if such error condition is met.